common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -617,6 +842,92 @@ public AccountRecord getAccountWithLocalAccountId(@Nullable final String environ
+                                                                        @NonNull final AccountRecord accountRecord) {
+        final OAuth2TokenCache cache = getTokenCacheForClient(
+                clientId,
+                accountRecord.getEnvironment(),
+                mCallingProcessUid
+        );
+
+        return cache.getAllTenantAccountsForAccountByClientId(
+                clientId,
+                accountRecord
+        );
+    }
+
+    @Override
+    public List<ICacheRecord> getAccountsWithAggregatedAccountData(@Nullable String environment,
+                                                                   @NonNull String clientId) {
+        final String methodName = ":getAccountsWithAggregatedAccountData";
+
+        final List<ICacheRecord> result;
+        OAuth2TokenCache targetCache;
+
+        if (null != environment) {
+            targetCache = getTokenCacheForClient(
+                    clientId,
+                    environment,
+                    mCallingProcessUid
+            );
+
+            if (null == targetCache) {
+                Logger.verbose(
+                        TAG + methodName,
+                        "Falling back to FoCI cache..."
+                );
+
+                targetCache = mFociCache;
+            }
+
+            result = targetCache.getAccountsWithAggregatedAccountData(environment, clientId);
+        } else {
+            // If no environment was specified, return all of the accounts across all of the envs...
+            // Callers should really specify an environment...
+            final List<OAuth2TokenCache> caches = getTokenCachesForClientId(clientId);
+
+            // Declare a new List to which we will add all of our results...
+            result = new ArrayList<>();
+
+            for (final OAuth2TokenCache cache : caches) {
+                result.addAll(cache.getAccountsWithAggregatedAccountData(environment, clientId));
+            }
+        }
+
+        return result;
+    }
+
+    @Override
+    public List<IdTokenRecord> getIdTokensForAccountRecord(@Nullable String clientId,
+                                                           @NonNull final AccountRecord accountRecord) {
+        List<IdTokenRecord> result;
+
+        final String accountEnv = accountRecord.getEnvironment();
+
+        if (null != clientId) {
+            final OAuth2TokenCache cache = getTokenCacheForClient(
+                    clientId,
+                    accountEnv,
+                    mCallingProcessUid
+            );
+
+            result = cache.getIdTokensForAccountRecord(
+                    clientId,
+                    accountRecord
+            );
+        } else {
+            // If the client id was null... then presumably we want to aggregate the IdTokens across
+            // apps... why would you want that? For now, throw an Exception and see if anyone requests
```

Comment:
Should we make it @NonNull in that case?

Reply:
Resolved in ad4e15dde76044b232e91be74420dad5e3aab778